File: mcp/src/minecraft/femtocraft/industry/gui/GuiEncoder.java
package femtocraft.industry.gui;

import femtocraft.Femtocraft;
import femtocraft.industry.containers.ContainerEncoder;
import femtocraft.industry.tiles.TileEntityEncoder;
import femtocraft.render.RenderUtils;
import femtocraft.utils.FemtocraftUtils;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.inventory.GuiContainer;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.util.Icon;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.StatCollector;
import net.minecraftforge.common.ForgeDirection;
import net.minecraftforge.fluids.FluidRegistry;
import net.minecraftforge.fluids.FluidStack;
import org.lwjgl.opengl.GL11;

public class GuiEncoder extends GuiContainer {

    public static final ResourceLocation texture = new ResourceLocation(
            Femtocraft.ID.toLowerCase(), "textures/guis/Encoder.png");
    private TileEntityEncoder encoder;

    public GuiEncoder(InventoryPlayer par1InventoryPlayer,
                      TileEntityEncoder par2Encoder) {
        super(new ContainerEncoder(par1InventoryPlayer, par2Encoder));
        this.encoder = par2Encoder;
    }

    /**
     * Draw the foreground layer for the GuiContainer (everything in front of
     * the items)
     */
    protected void drawGuiContainerForegroundLayer(int par1, int par2) {
        String s = "Encoder";
        this.fontRenderer.drawString(s,
                                     this.xSize / 2 - this.fontRenderer.getStringWidth(s) / 2, 6,
                                     FemtocraftUtils.colorFromARGB(0, 255, 255, 255));
        this.fontRenderer.drawString(
                StatCollector.translateToLocal("container.inventory"), 8,
                this.ySize - 96 + 2,
                FemtocraftUtils.colorFromARGB(0, 255, 255, 255));
    }

    /**
     * Draw the background layer for the GuiContainer (everything behind the
     * items)
     */
    protected void drawGuiContainerBackgroundLayer(float par1, int par2,
                                                   int par3) {
        GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
        Minecraft.getMinecraft().getTextureManager().bindTexture(texture);
        int k = (this.width - this.xSize) / 2;
        int l = (this.height - this.ySize) / 2;
        this.drawTexturedModalRect(k, l, 0, 0, this.xSize, this.ySize);
        int i1;

        // if (this.furnaceInventory.isBurning())
        // {
        // i1 = this.furnaceInventory.getBurnTimeRemainingScaled(12);
        // this.drawTexturedModalRect(k + 56, l + 36 + 12 - i1, 176, 12 - i1,
        // 14, i1 + 2);
        // }

        i1 = this.encoder.getProgressScaled(20);
        this.drawTexturedModalRect(k + 119, l + 25, 176, 0, 18, i1);
        i1 = (int) (this.encoder.getFillPercentage() * 60);
        this.drawTexturedModalRect(k + 10, l + 68 - i1, 176, 100 - i1, 16, i1);

        FluidStack fluid = this.encoder.getTankInfo(ForgeDirection.UNKNOWN)[0].fluid;
        if (fluid != null) {
            Icon image = fluid.getFluid().getStillIcon();
            // Icon image = BlockFluid.getFluidIcon("water_still");
            // image = Femtocraft.mass_block.stillIcon;

            i1 = (this.encoder.getMassAmount() * 60)
                    / this.encoder.getMassCapacity();
            RenderUtils.renderLiquidInGUI(this, this.zLevel, image, k + 150, l
                    + 8 + (60 - i1), 16, i1);

            // Rebind texture
            Minecraft.getMinecraft().getTextureManager().bindTexture(texture);
        }

        // Draw Tank Lines
        this.drawTexturedModalRect(k + 150, l + 8, 176, 100, 16, 60);
    }

    /*
     * (non-Javadoc)
     *
     * @see net.minecraft.client.gui.inventory.GuiContainer#drawScreen(int, int,
     * float)
     */
    @Override
    public void drawScreen(int par1, int par2, float par3) {
        super.drawScreen(par1, par2, par3);

        if (this.isPointInRegion(18, 12, 16, 60, par1, par2)) {
            int furnaceCurrent = this.encoder.getCurrentPower();
            int furnaceMax = this.encoder.getMaxPower();

            // String text = String.format("%i/%i", furnaceCurrent, furnaceMax);
            String text = String.valueOf(furnaceCurrent) + '/'
                    + String.valueOf(furnaceMax);

            this.drawCreativeTabHoveringText(text, par1, par2);

        }
        else if (this.isPointInRegion(150, 8, 16, 60, par1, par2)) {
            int massCurrent = this.encoder.getMassAmount();
            int massMax = this.encoder.getMassCapacity();

            FluidStack fluid = this.encoder.getTankInfo(ForgeDirection.UNKNOWN)[0].fluid;
            String name = fluid == null ? "" : (" " + FluidRegistry
                    .getFluidName(fluid));
            String text = String.valueOf(massCurrent) + '/'
                    + String.valueOf(massMax) + " mB" + name;

            this.drawCreativeTabHoveringText(text, par1, par2);
        }
    }
}

File: src/main/java/com/itszuvalex/femtocraft/industry/gui/GuiEncoder.scala
/*
 * ******************************************************************************
 *  * Copyright (C) 2013  Christopher Harris (Itszuvalex)
 *  * Itszuvalex@gmail.com
 *  *
 *  * This program is free software; you can redistribute it and/or
 *  * modify it under the terms of the GNU General Public License
 *  * as published by the Free Software Foundation; either version 2
 *  * of the License, or (at your option) any later version.
 *  *
 *  * This program is distributed in the hope that it will be useful,
 *  * but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  * GNU General Public License for more details.
 *  *
 *  * You should have received a copy of the GNU General Public License
 *  * along with this program; if not, write to the Free Software
 *  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 *  *****************************************************************************
 */
package com.itszuvalex.femtocraft.industry.gui

import com.itszuvalex.femtocraft.Femtocraft
import com.itszuvalex.femtocraft.core.gui.GuiBase
import com.itszuvalex.femtocraft.industry.containers.ContainerEncoder
import com.itszuvalex.femtocraft.industry.gui.GuiEncoder._
import com.itszuvalex.femtocraft.industry.tiles.TileEntityEncoder
import com.itszuvalex.femtocraft.render.RenderUtils
import com.itszuvalex.femtocraft.utils.FemtocraftUtils
import cpw.mods.fml.relauncher.{Side, SideOnly}
import net.minecraft.client.Minecraft
import net.minecraft.entity.player.{EntityPlayer, InventoryPlayer}
import net.minecraft.util.{ResourceLocation, StatCollector}
import net.minecraftforge.common.util.ForgeDirection
import net.minecraftforge.fluids.FluidRegistry
import org.lwjgl.opengl.GL11

@SideOnly(Side.CLIENT) object GuiEncoder {
  val texture = new ResourceLocation(Femtocraft.ID.toLowerCase, "textures/guis/Encoder.png")
}

@SideOnly(Side.CLIENT) class GuiEncoder(player: EntityPlayer, par1InventoryPlayer: InventoryPlayer, private val encoder: TileEntityEncoder) extends GuiBase(new ContainerEncoder(player, par1InventoryPlayer, encoder)) {

  override def drawScreen(par1: Int, par2: Int, par3: Float) {
    super.drawScreen(par1, par2, par3)
    if (isPointInRegion(18, 12, 16, 60, par1, par2)) {
      val furnaceCurrent = encoder.getCurrentPower
      val furnaceMax = encoder.getMaxPower
      val text = FemtocraftUtils.formatIntegerToString(furnaceCurrent) + '/' + FemtocraftUtils.formatIntegerToString(furnaceMax)
      drawCreativeTabHoveringText(text, par1, par2)
    }
    else if (isPointInRegion(150, 8, 16, 60, par1, par2)) {
      val massCurrent = encoder.getMassAmount
      val massMax = encoder.getMassCapacity
      val fluid = encoder.getTankInfo(ForgeDirection.UNKNOWN)(0).fluid
      val name = if (fluid == null) "" else " " + FluidRegistry.getFluidName(fluid)
      val text = FemtocraftUtils.formatIntegerToString(massCurrent) + '/' + FemtocraftUtils.formatIntegerToString(massMax) + " mB" + name
      drawCreativeTabHoveringText(text, par1, par2)
    }
  }

  /**
   * Draw the foreground layer for the GuiContainer (everything in front of the items)
   */
  protected override def drawGuiContainerForegroundLayer(par1: Int, par2: Int) {
    val s = "Encoder"
    fontRendererObj.drawString(s, xSize / 2 - fontRendererObj.getStringWidth(s) / 2, 6, FemtocraftUtils.colorFromARGB(0, 255, 255, 255))
    fontRendererObj.drawString(StatCollector.translateToLocal("container.inventory"), 8, ySize - 96 + 2, FemtocraftUtils.colorFromARGB(0, 255, 255, 255))
  }

  /**
   * Draw the background layer for the GuiContainer (everything behind the items)
   */
  protected def drawGuiContainerBackgroundLayer(par1: Float, par2: Int, par3: Int) {
    GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F)
    Minecraft.getMinecraft.getTextureManager.bindTexture(texture)
    val k = (width - xSize) / 2
    val l = (height - ySize) / 2
    drawTexturedModalRect(k, l, 0, 0, xSize, ySize)
    var i1 = 0
    i1 = encoder.getProgressScaled(20)
    drawTexturedModalRect(k + 119, l + 25, 176, 0, 18, i1)
    i1 = (encoder.getFillPercentage * 60).toInt
    drawTexturedModalRect(k + 10, l + 68 - i1, 176, 100 - i1, 16, i1)
    val fluid = encoder.getTankInfo(ForgeDirection.UNKNOWN)(0).fluid
    if (fluid != null) {
      val image = fluid.getFluid.getStillIcon
      i1 = (encoder.getMassAmount * 60) / encoder.getMassCapacity
      RenderUtils.renderLiquidInGUI(this, zLevel, image, k + 150, l + 8 + (60 - i1), 16, i1)
      Minecraft.getMinecraft.getTextureManager.bindTexture(texture)
    }
    drawTexturedModalRect(k + 150, l + 8, 176, 100, 16, 60)
  }
}